src/Microsoft.Azure.SignalR.Common/Exceptions/AzureSignalRAccessTokenNotAuthorizedException.cs

```diff
@@ -36,17 +38,17 @@ public AzureSignalRAccessTokenNotAuthorizedException(string credentialName, Exce
     /// <summary>
     /// Initializes a new instance of the <see cref="AzureSignalRAccessTokenNotAuthorizedException"/> class.
     /// </summary>
-    internal AzureSignalRAccessTokenNotAuthorizedException(TokenCredential credential, Exception inner) :
+    internal AzureSignalRAccessTokenNotAuthorizedException(TokenCredential credential, Exception? inner) :
         base(string.Format(Template, credential.GetType().Name, GetInnerReason(inner)), inner)
     {
     }
 
-    private static string GetInnerReason(Exception exception)
+    private static string GetInnerReason(Exception? exception)
     {
         return exception switch
         {
             AzureSignalRUnauthorizedException => AzureSignalRUnauthorizedException.ErrorMessageMicrosoftEntra,
-            _ => exception.Message,
+            _ => exception?.Message ?? "The access key has expired.",
```

Comment:
this might be correct for now, but making this a default assumption in this exception class sounds not an extensible solution